dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/tracking/AndroidXFragmentLifecycleCallbacks.kt

```diff
@@ -55,18 +71,44 @@ internal class AndroidXFragmentLifecycleCallbacks(
     override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
         super.onFragmentResumed(fm, f)
         componentPredicate.runIfValid(f) {
-            GlobalRum.get()
-                .startView(
-                    it,
-                    it.resolveViewName(),
-                    argumentsProvider(it)
+            viewLoadingTimer.onFinishedLoading(resolveKey(it))
+            rumEventListener.onStartView(
```

Comment:
Why are you adding a level of abstraction here ? Is it really necessary ? Why not just make the "onStartView" / "onStopView" methods open in this class and let the `NavControllerFragmentLifecycleCallbacks` override them to **not** log anything.

Reply:
Wanted to delegate the the stop/pause/updateTime to the strategy. I did this mainly to extract that common code from AndroidXLifecycleCallback and OreoLifecycleCallback. I am not sure which way is better, if I go with the opened methods inside the Callback and drop the delegate that means I will have to have this duplicated code in those callbacks. What do you think ?

Reply:
I feel that I am running in circles...before I had those methods/classes abstract and implemented those when I instantiate the objects inside the strategies...then I switch to this one, not sure which one is better :).

Reply:
I'll still have to extract the Base class from those callbacks...but yeah...maybe is better.